.github/workflows/tests.yml

```diff
@@ -445,6 +445,159 @@ jobs:
+          export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LXD_DOCUMENTATION="/home/runner/work/lxd/lxd/doc/_build/"
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo rm -rf /var/lib/lxd
+          sudo -E PATH=${PATH} LD_LIBRARY_PATH=${LD_LIBRARY_PATH} $(go env GOPATH)/bin/lxd --group sudo &
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: 'canonical/lxd-ui'
+          ref: main
+          path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Install LXD-UI dependencies
+        run: |
+          set -eux
+          sudo chmod 0777 ./lxd-ui
+          cd lxd-ui
+          dotrun install
+
+      - name: Run LXD-UI
+        env:
+          ENVIRONMENT: devel
+          PORT: 8407
+          LXD_UI_BACKEND_IP: 172.17.0.1
+        run: |
+          set -eux
+          cd lxd-ui
+          dotrun &
+          curl --head --fail --retry-delay 2 --retry 100 --retry-connrefused --insecure https://localhost:${{ env.PORT }}
+
+      - name: Setup LXD
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc storage create default zfs
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default root disk path=/ pool=default
+          sudo -E LXD_DIR=/var/lib/lxd lxc network create local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default eth0 nic network=local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set core.https_address "[::]:8443"
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set cluster.https_address "127.0.0.1"
+          sudo -E LXD_DIR=/var/lib/lxd lxc cluster enable local
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set user.show_permissions=true
+          sudo -E LXD_DIR=/var/lib/lxd lxc config trust add lxd-ui/keys/lxd-ui.crt
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install Playwright Browsers
+        run: |
+          set -eux
+          cd lxd-ui
+          npx playwright install --with-deps chromium
+
+      - name: Setup for LXD-UI tests
+        shell: bash
+        run: |
+          set -eux
+          sudo -E LXD_DIR=/var/lib/lxd PATH=/home/runner/go/bin:$PATH ./lxd-ui/tests/scripts/setup_test
+
+      - name: Run Playwright tests
+        run: |
+          set -eux
+          cd lxd-ui
+          sudo chown $USER -R /home/runner/.config
+          sudo chown $USER /var/lib/lxd/unix.socket
+          CI=true DISABLE_VM_TESTS=true LXD_DIR=/var/lib/lxd PATH=/home/runner/go/bin:$PATH npx playwright test --project chromium:lxd-latest-edge
+          #npx playwright test --project chromium:lxd-5.21-edge
+          #npx playwright test --project lxd-5.0-edge
+
```

Comment:
We should probably have a step here to cleanly stop LXD via `lxd shutdown` so we can be sure all coverage data is written before sending it up.

Reply:
I added the `lxd shutdown`, but the coverage data for the ui is empty on [this run](https://github.com/edlerd/lxd/actions/runs/11958693042/job/33339373448) on my fork. I set up the secrets over there and the e2e test suite completed just fine. Any idea what could be missing is welcome.

Reply:
I wonder if the GOCOVERDIR env var isn't being set in your fork's run because of this line

https://github.com/canonical/lxd/blob/5fdd983438347b63cb4077080ba4c64548a8e8a5/.github/workflows/tests.yml#L15

Reply:
> I wonder if the GOCOVERDIR env var isn't being set in your fork's run because of this line

Yeah, I saw that and explicitly set `GOCOVERDIR` for the [ui step in recent versions](https://github.com/canonical/lxd/pull/14035/files#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR454) of the workflow.

Reply:
[I'd suggest ](https://github.com/canonical/lxd/blob/5fdd983438347b63cb4077080ba4c64548a8e8a5/.github/workflows/tests.yml#L122) then relies on that env var being set here:

https://github.com/canonical/lxd/blob/main/Makefile#L39

Reply:
This [recent run](https://github.com/edlerd/lxd/actions/runs/12030477528) on my fork produced coverage data for the ui run. I am not sure what caused this before to fail though, but happy that it works now :)